app/src/main/java/ch/epfl/balelecbud/EmergencyNumbersActivity.java

```diff
@@ -0,0 +1,123 @@
+package ch.epfl.balelecbud;
+
+import android.Manifest;
+import android.content.Intent;
+import android.content.pm.PackageManager;
+import android.net.Uri;
+import android.os.Bundle;
+import android.util.Log;
+import android.widget.ArrayAdapter;
+import android.widget.ListView;
+
+import androidx.core.app.ActivityCompat;
+
+import com.google.firebase.firestore.CollectionReference;
+import com.google.firebase.firestore.FirebaseFirestore;
+import com.google.firebase.firestore.QueryDocumentSnapshot;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import ch.epfl.balelecbud.emergency.models.EmergencyNumber;
+import ch.epfl.balelecbud.util.database.DatabaseWrapper;
+
+import static ch.epfl.balelecbud.BalelecbudApplication.getAppDatabaseWrapper;
+
+public class EmergencyNumbersActivity extends BasicActivity {
+    public static final int PERMISSION_TO_CALL_CODE = 991;
+    private boolean callPermissionGranted;
+    private ListView numbersListView;
+    private ArrayAdapter arrayAdapter;
+    private Map<String, String> repertoryMap = new HashMap<String, String>();
+
+    private FirebaseFirestore db = FirebaseFirestore.getInstance();
+    private CollectionReference emergencyNumbersRef = db.collection("emergencyNumbers");
+
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        BalelecbudApplication.getAppDatabaseWrapper();
+
+        setContentView(R.layout.activity_emergency_numbers);
+        configureToolBar(R.id.emergency_numbers_activity_toolbar);
+        configureDrawerLayout(R.id.emergency_numbers_activity_drawer_layout);
+        configureNavigationView(R.id.emergency_numbers_activity_nav_view);
+
+
+        numbersListView = findViewById(R.id.numbersListView);
+        arrayAdapter = new ArrayAdapter(
+                this,
+                android.R.layout.simple_list_item_1,
+                new ArrayList<String>());
+    }
+
+    @Override
+    protected void onStart() {
+        super.onStart();
+        emergencyNumbersRef.addSnapshotListener(this, (queryDocumentSnapshots, e) -> {
```

Comment:
You shouldn't expose the database implementation like this. Use the method `query` from the database wrapper, to which you can pass a query of the style `MyQuery(DatabaseWrapper. EMERGENCY_NUMBER_PATH, new LinkedList<>())` and the type `EmergencyNumber.class`